1-Docker/headnode_dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM alpine:3.12
+
+RUN apk add \
+    openjdk8 \
+    openssh \
+    bash \
+    sudo \
+    curl
+
+RUN addgroup hadoop && \
+    adduser -DG hadoop hadoop && \
+    adduser -DG hadoop yarn && \
+    adduser -DG hadoop hdfs
+
+RUN curl -O https://archive.apache.org/dist/hadoop/common/hadoop-3.1.2/hadoop-3.1.2.tar.gz && \
+    tar -zxf hadoop-3.1.2.tar.gz -C /usr/local/ && rm hadoop-3.1.2.tar.gz
+
+VOLUME /opt/namenode-dir
+
+ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk/jre
+ENV HADOOP_HOME /usr/local/hadoop-3.1.2
+
+COPY scripts/set-hadoop-conf.sh /usr/local/set-hadoop-conf.sh
+
+RUN /usr/local/set-hadoop-conf.sh
+
+RUN chown root:hadoop $HADOOP_HOME && \
+    chown root:hadoop $HADOOP_HOME/* && \
```

Comment:
what is the expected result after that command?

Reply:
*/usr/local/set-hadoop-conf.sh* - this script 
downloads hadoop config files (hadoop-env.sh, core-site.xml, hdfs-site.xml, yarn-site.xml) - the same that we had on Linux Exam - and set required parameters (JAVA_HOME, HADOOP_HOME, hostnames and dirs)

As we untared hadoop archive under root, *I needed to change the owning group of the hadoop working files and directories (to escape Permissiom denied fault)*

So in these lines I change the owning user and group of $HADOOP_HOME directory recursively (when I tested these commands, somehow **-r** parameter didn't work, so I put it in two lines)

Reply:
1. Why didn't "chown -r" work?
2. Does this command changes owner for **all** files and directories inside $HADOOP_HOME/? 
chown root:hadoop $HADOOP_HOME/*

Reply:
> Why didn't "chown -r" work?

*/bin/sh* doesn't support **-r** parameter. I just checked its "help page", it does support **-R** parameter :'D 
So I really had a chance to make it one-line comand 

> Does this command changes owner for all files and directories inside $HADOOP_HOME/?
chown root:hadoop $HADOOP_HOME/*

No, it apply changes only for **1-st layer** of $HADOOP_HOME dir. Using **-R** would do the job, but it wasn't required :)

Hadoop was complaing on having lack of permissions for writing in home dir. Now I'm thinking that change owner of $HADOOP_HOME dir and $HADOOP_HOME/logs (after fomatting of FS) later  would be just enough